### PR TITLE
adding java check before starting server #11

### PR DESCRIPTION
**Pull request by abdrabrcom**
Signed-off-by: Ramy Abdalla <ramy.abdalla@broadcom.com>

Applied changes to check for java installation before launching LSP server

### CODE REVIEW

File: clients/cobol-lsp-vscode-extension/src/extension.ts [hunk 1]
```diff
@@ -12,15 +12,17 @@
...
  */
 
+import * as cp from "child_process";
 import * as net from "net";
 import {ExtensionContext, extensions, StatusBarAlignment, window} from "vscode";
 import {
     Executable,
     LanguageClient,
     LanguageClientOptions,
+    StreamInfo,
```
Comment (shciubrcom) [inline clients/cobol-lsp-vscode-extension/src/extension.ts]: StreamInfo is never used

**Comment by abdrabrcom** [comment]
@shciubrcom removed unneaded imports

@zacanbrcom please check fixed implementation

File: clients/cobol-lsp-vscode-extension/src/extension.ts [hunk 2]
```diff
@@ -60,3 +61,26 @@ export function activate(context: ExtensionContext) {
...
+        });
+        ls.on("error", (code: any) => {
+            if ("Error: spawn java ENOENT" === code.toString()) {
+                reject("Java 8 is not found");
+            }
+            reject(code);
+        });
+        ls.on("close", (code: number) => {
+            if (code !== 0) {
+                reject("An error occured when checking if Java was installed");
```
Comment (zacanbrcom) [inline clients/cobol-lsp-vscode-extension/src/extension.ts]: pls fix the typo.. I checked the overall behaviour and it works fine!

Comment (abdrabrcom) [reply clients/cobol-lsp-vscode-extension/src/extension.ts]: Typo fixed, change has been pushed in the repository

**Pull request merged**